src/android/com/adobe/phonegap/push/GCMIntentService.java

```diff
@@ -87,8 +87,12 @@ protected void onMessage(Context context, Intent intent) {
         // Extract the payload from the message
         Bundle extras = intent.getExtras();
         if (extras != null) {
+
+            SharedPreferences prefs = context.getSharedPreferences(PushPlugin.COM_ADOBE_PHONEGAP_PUSH, Context.MODE_PRIVATE);
+            boolean forceShow = prefs.getBoolean(FORCE_SHOW, false);
+
             // if we are in the foreground, just surface the payload, else post it to the statusbar
-            if (PushPlugin.isInForeground()) {
+            if (!forceShow && PushPlugin.isInForeground()) {
```

Comment:
I'm not sure of the practical cases, but in theory one could send a push notification with `message === null` and still want it to be shown with something else (like a big picture, or a summaryText, or whatever else he might provide other than message).